pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go

```diff
@@ -112,6 +112,13 @@ func (a *Authorizer) HasCapabilityForBatch(
 		if entry.ServiceMode == mtinfopb.ServiceModeNone || entry.ServiceMode == mtinfopb.ServiceModeStopping {
 			return errors.Newf("operation not allowed when in service mode %q", entry.ServiceMode)
 		}
+		// Shared service tenants in UA implicitly have all capabilities. If/when
+		// we offer shared service for truly _multi-tenant_ deployments and wish to
+		// restrict some of those tenants, we can add another service mode that is
+		// similar to shared but adds restriction to only granted capabilities.
+		if entry.ServiceMode == mtinfopb.ServiceModeShared {
+			return nil
+		}
```

Comment:
Do we also want to do this in the other capability reader methods such as `HasNodeStatusCapability`. If so, perhaps we want to reorganize the code a bit to make that easier.

Reply:
made `getMode` just select allowAll based on svcmode instead of needing to update the auth-on case of every method, since they're all already obligated to check the mode.